superchain/go.mod

```diff
@@ -1,12 +1,54 @@
 module github.com/ethereum-optimism/superchain-registry/superchain
 
-go 1.20
-
+go 1.21
 require (
-	golang.org/x/crypto v0.13.0
-	golang.org/x/exp v0.0.0-20230905200255-921286631fa9
-	golang.org/x/mod v0.12.0
+	github.com/ethereum/go-ethereum v1.13.10
+	golang.org/x/crypto v0.18.0
+	golang.org/x/exp v0.0.0-20231110203233-9a3e6036ecaa
+	golang.org/x/mod v0.14.0
```

Comment:
So far we have avoided depending on any Implementation. Especially upstream geth. Since the registry gets pulled in by the monorepo, which depends on a different conflicting version of geth.

If we build tooling to check advanced things in this repo, it would be better to put it as CLI tool in the optimism monorepo, and then call the CLI from the CI here.

Reply:
Thanks for the feedback @protolambda! I hadn't considered the conflicting geth versions. I think we could remove the dependency on geth. We are only making a very simple on chain query, and that could be done with a hand rolled JSON-RPC request.

Reply:
For full context:

My instinct is to steer away from using CLI tools from the monorepo in this repo. That's for two reasons:
1. Running compiled Go code as an executable, and then parsing the output (probably with more Go code) in order to inspect / assert on it seems somewhat inelegant and brittle (no type checking, for example).
2. This would bake in a cyclic dependency -- not one the Go compiler is going to complain about, but still somewhat of an antipattern.